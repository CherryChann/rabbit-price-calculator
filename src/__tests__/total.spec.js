import React from 'react';
import { shallow, mount } from 'enzyme';
import Total from '../components/total';
import { addDays } from 'date-fns';

const onClick = (value) => { 
    console.log(value, 'mock function');
}
const props = {
    totalCost: 3480,
    totalUnits: 1200,
    product: {
        id: 1,
        name: "Flyer - One Sided",
        max_production: {
            "1": 5000,
            "2": 8000,
            "3": 12000
        },
        price_per_unit: 0.01
    },
    days: 4,
    date: addDays(new Date(), 4),
    onClick: onClick
}

describe('Shallow rendered total component ', () => {
    const container = shallow(
        <Total 
            totalCost={props.totalCost}
            totalUnits={props.totalUnits} 
            days={props.days}
            date={props.date}
            product={props.product}
        ></Total>);
    it('should render select box with products', () => {
        expect(container.html()).toMatchSnapshot();
    });

    it('should have correct props', () => {
        let totalUnit = container.find('#total-unit-badge')
        expect(totalUnit.text()).toEqual("1200")
        let totalCost = container.find('#total-cost-badge')
        expect(totalCost.text()).toEqual("3480")
    })

    it('should show the alert box if quantity is more than max production of product according to date', () => {
        const container = shallow(
            <Total 
                totalCost={10000}
                totalUnits={100000} 
                days={props.days}
                date={props.date}
                product={props.product}
            ></Total>);
        expect(container.find('#total-units-error').length).toEqual(1);
    })
});