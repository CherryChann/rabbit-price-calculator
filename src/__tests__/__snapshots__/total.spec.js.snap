// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow rendered total component  should render total component 1`] = `"<div><div class=\\"margin-top row\\"><div class=\\"form-group total-unit-padding-top col-lg-2 col-5\\"><span>Total Units: </span></div><div class=\\"col-lg-3 col-7\\"><h4><span id=\\"total-unit-badge\\" class=\\"badge-vertical-align badge badge-secondary\\"><strong>1200</strong></span></h4></div></div><div class=\\"margin-top row\\"><div class=\\"form-group col-lg-2 col-5\\"><span>Total Cost:</span></div><div class=\\"col-lg-3 col-7\\"><h4><span id=\\"total-cost-badge\\" class=\\"badge-vertical-align badge badge-secondary\\"><strong>3480</strong></span></h4></div></div></div>"`;
